toqito/state_props/tests/test_is_unextendible_product_basis.py

```diff
@@ -0,0 +1,68 @@
+"""Test is_unextendible_product_basis."""
+import numpy as np
+
+from toqito.matrix_ops import tensor
+from toqito.state_props import is_unextendible_product_basis
+from toqito.states import basis, bell, tile
+
+
+def test_is_unextensible_product_state_non_product_state():
+    """Check if exception raised if non-product state is passed."""
+    bell_states = [bell(i) for i in range(4)]
+    dims = [2, 2]
+    np.testing.assert_raises(ValueError, is_unextendible_product_basis, bell_states, dims)
+
+
+def test_is_unextensible_product_state_wrong_size():
+    """Check if exception raised if size of a vector does not match product of dims."""
+    bell_states = [bell(i) for i in range(4)]
+    dims = [2, 3]
+    np.testing.assert_raises(ValueError, is_unextendible_product_basis, bell_states, dims)
```

Comment:
You could condense this into 1 test using `pytest.parameterize`. 

Here's one example of it.

https://github.com/vprusso/toqito/blob/8ec4bf39167580d5f62488c7d73a9929aa1f43d7/toqito/channels/tests/test_partial_trace.py#L12-L20